mkctr.go

```diff
@@ -198,6 +203,34 @@ func verifyPlatform(p v1.Platform, target string) error {
 	return nil
 }
 
+func createOutDirectory(path string) error {
+	fi, err := os.Stat(path)
```

Comment:
(drive-by) Are the stat and error and type checks necessary here? `os.MkdirAll` already handles the case where the directory exists, and correctly errors when the path exists but is _not_ a directory. https://go.dev/play/p/2C56cHeDKBV